textdb/textdb-dataflow/src/main/java/edu/uci/ics/textdb/dataflow/keywordmatch/KeywordMatcher.java

```diff
@@ -0,0 +1,153 @@
+            e.printStackTrace();
+            throw new DataFlowException(e.getMessage(), e);
+        }
+    }
+
+    @Override
+    public ITuple getNextTuple() throws DataFlowException {
+
+        List<IField> fieldList;
+        boolean foundFlag = false;
+        int positionIndex = 0; // Next position in the field to be checked.
+        int spanIndexValue; // Starting position of the matched query
+
+        try {
+            ITuple sourceTuple = sourceOperator.getNextTuple();
+            if(sourceTuple == null){
+                return null;
+            }
+            fieldList = sourceTuple.getFields();
+            spanList.clear();
+            if(!schemaDefined){
+                schemaDefined = true;
+                Schema schema = sourceTuple.getSchema();
+                spanSchema = Utils.createSpanSchema(schema);
+            }
+
+            for(int attributeIndex = 0; attributeIndex < attributeList.size(); attributeIndex++){
+                IField field = sourceTuple.getField(attributeList.get(attributeIndex).getFieldName());
+                String fieldValue = (String) (field).getValue();
+                String fieldName;
+                if(field instanceof StringField){
+                    //Keyword should match fieldValue entirely
+                    if(fieldValue.equals(query.toLowerCase())){
+                        spanIndexValue = 0;
+                        positionIndex = query.length();
+                        fieldName = attributeList.get(attributeIndex).getFieldName();
+                        addSpanToSpanList(fieldName, spanIndexValue, positionIndex, query, fieldValue);
+                        foundFlag = true;
+                    }
+                }
+                else if(field instanceof TextField) {
+                    String documentValue;
+                    Matcher matcher;
+                    //Each element of Array of keywords is matched in tokenized TextField Value
+                    for(int iter = 0; iter < queryTokens.size(); iter++) {
+                        positionIndex = 0;
+                        String query = queryTokens.get(iter);
+                        //Ex: For keyword lin it obtains pattern like /blin/b which matches keywords at boundary
+                        Pattern pattern = tokenPatternList.get(iter);
+                        matcher = pattern.matcher(fieldValue.toLowerCase());
+                        while (matcher.find(positionIndex) != false) {
+                            spanIndexValue = matcher.start();
+                            positionIndex = spanIndexValue + queryTokens.get(iter).length();
+                            documentValue = fieldValue.substring(spanIndexValue, positionIndex);
+                            fieldName = attributeList.get(attributeIndex).getFieldName();
+                            addSpanToSpanList(fieldName, spanIndexValue, positionIndex, query, documentValue);
+                            foundFlag = true;
```

Comment:
@chenlica .. I was of the opinion that, we should follow "OR" logic for `KeyWordMatcher`, since we are dealing with the keyTokens separately and we are representing them separately in spans. 
But we can implement the "AND" logic, if that is preferable.